client/my-sites/email/email-providers-comparison/index.jsx

```diff
@@ -298,13 +299,29 @@ class EmailProvidersComparison extends Component {
 			} );
 	};
 
+	onSuccessAddForwarding = () => {
+		const { domain, getSiteDomains, requestingSiteDomains, selectedSite } = this.props;
+		if ( ! requestingSiteDomains ) {
+			getSiteDomains( selectedSite.ID );
+		}
```

Comment:
For now, I think this is OK.

I just took a closer look and the issue seems to be that we're not adding a `<QuerySiteDomains />` component to the content we render in `EmailManagementHome` for the two code paths where we end up rendering `EmailProvidersComparison`. But we can take that on in a follow-up PR.